### PR TITLE
rgw: add bucket as option when show/trim usage

### PR DESCRIPTION
**Pull request by dongbula**
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by dongbula** [comment]
@cbodley hi, can we continue this pr?

File: src/cls/rgw/cls_rgw.cc [hunk 1]
```diff
@@ -2930,7 +2930,7 @@ int rgw_user_usage_log_add(cls_method_context_t hctx, bufferlist *in, bufferlist
 }
 
 static int usage_iterate_range(cls_method_context_t hctx, uint64_t start, uint64_t end,
-                            string& user, string& key_iter, uint32_t max_entries, bool *truncated,
+                            string& user, string& bucket, string& key_iter, uint32_t max_entries, bool *truncated,
```
Comment (cbodley) [inline src/cls/rgw/cls_rgw.cc]: please pass bucket by `const string&` (here and elsewhere)

File: src/test/cls_rgw/test_cls_rgw.cc [hunk 2]
```diff
@@ -825,22 +835,42 @@ TEST(cls_rgw, usage_basic)
...
+  string bucket1 = "bucket-usage-1";
+  string bucket2 = "bucket-usage-2";
+  info = gen_usage_log_info(user, payer, bucket1, 100);
+  cls_rgw_usage_log_add(op, info);
+  info = gen_usage_log_info(user, payer, bucket2, 100);
+  cls_rgw_usage_log_add(op, info);
+  ret = cls_rgw_usage_log_read(ioctx, oid, user, bucket1, start_epoch, end_epoch,
+                              max_entries, read_iter, usage2, &truncated);
+  ASSERT_EQ(0, ret);
+  ASSERT_EQ(100, usage2.size());
```
Comment (cbodley) [inline src/test/cls_rgw/test_cls_rgw.cc]: i'm seeing this fail in testing:
```
2018-09-11T19:00:20.885 INFO:tasks.workunit.client.0.smithi167.stdout:[ RUN      ] cls_rgw.usage_basic
2018-09-11T19:00:26.822 INFO:tasks.workunit.client.0.smithi167.stdout:/src/test/cls_rgw/test_cls_rgw.cc:864: Failure
2018-09-11T19:00:26.822 INFO:tasks.workunit.client.0.smithi167.stdout:      Expected: 100
2018-09-11T19:00:26.822 INFO:tasks.workunit.client.0.smithi167.stdout:To be equal to: usage2.size()
2018-09-11T19:00:26.822 INFO:tasks.workunit.client.0.smithi167.stdout:      Which is: 0
```

**Comment by cbodley** [comment]
hi @dongbula, i'm still seeing failures here - this time:
```
2018-09-16T13:49:07.302 INFO:tasks.workunit.client.0.smithi097.stdout:/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos7/DIST/centos7/MACHINE_SIZE/huge/release/14.0.0-3280-g4a18582/rpm/el7/BUILD/ceph-14.0.0-3280-g4a18582/src/test/cls_rgw/test_cls_rgw.cc:867: Failure
2018-09-16T13:49:07.303 INFO:tasks.workunit.client.0.smithi097.stdout:      Expected: 100
2018-09-16T13:49:07.303 INFO:tasks.workunit.client.0.smithi097.stdout:To be equal to: usage2.size()
2018-09-16T13:49:07.303 INFO:tasks.workunit.client.0.smithi097.stdout:      Which is: 1
```

**Comment by yuriw** [comment]
[wip-yuri5-testing-2018-09-24-1540](https://trello.com/c/xPnTXigg/410-wip-yuri5-testing-2018-09-24-1540)

**Comment by dongbula** [comment]
@cbodley hi, please test it again, it have repaired and tested it locally, thanks

**Comment by cbodley** [comment]
@yuriw, i see that your test results are showing the test_cls_rgw failures from this pr - could you please pull in this update and rerun?

**Comment by yuriw** [comment]
[wip-yuri2-testing-2018-11-15-2111](https://trello.com/c/NmUyiuZp/472-wip-yuri2-testing-2018-11-15-2111)

**Comment by yuriw** [comment]
> @yuriw, i see that your test results are showing the test_cls_rgw failures from this pr - could you please pull in this update and rerun?

Yes, what your email I'll add you to trello

**Pull request merged**